staging/src/k8s.io/client-go/tools/remotecommand/remotecommand_test.go

```diff
@@ -264,3 +300,65 @@ func v4WriteStatusFunc(stream io.Writer) func(status *apierrors.StatusError) err
 		return err
 	}
 }
+
+func newStreamExecutor(config *rest.Config, method string, url *url.URL) (*streamExecutor, error) {
+	wrapper, upgradeRoundTripper, err := spdytransport.RoundTripperFor(config)
+	if err != nil {
+		return nil, err
+	}
+	return &streamExecutor{
+		upgrader:  upgradeRoundTripper,
+		transport: wrapper,
+		method:    method,
+		url:       url,
+		protocols: []string{
+			remotecommandconsts.StreamProtocolV4Name,
+			remotecommandconsts.StreamProtocolV3Name,
+			remotecommandconsts.StreamProtocolV2Name,
+			remotecommandconsts.StreamProtocolV1Name,
+		},
+	}, nil
+}
+
+func TestStreamExitsAfterConnectionIsClosed(t *testing.T) {
+	options := StreamOptions{
+		Stdin:  &fakeEmptyDataPty{},
+		Stdout: &fakeEmptyDataPty{},
+	}
+	server := newTestHTTPServer(fakeMassiveDataAttacher, &options)
+
+	ctx, cancelFn := context.WithTimeout(context.Background(), 500*time.Millisecond)
+	defer cancelFn()
+
+	uri, _ := url.Parse(server.URL)
+	exec, err := newStreamExecutor(&rest.Config{Host: uri.Host}, "POST", uri)
```

Comment:
Why not just using the real `NewSPDYExecutor` with a type assertion?

Reply:
Clever way. I haven't thought of that.